presto-docs/src/main/sphinx/functions/json.rst

```diff
@@ -1,6 +1,65 @@
-==============
-JSON Functions
-==============
+============================
+JSON Functions And Operators
+============================
+
+JSON operators
+--------------
+
+.. function:: Cast to JSON
+
+    Cast from ``BOOLEAN``, ``TINYINT``, ``SMALLINT``, ``INTEGER``,
+    ``BIGINT``, ``REAL``, ``DOUBLE``, ``VARCHAR`` to ``JSON`` is supported.
+    Cast from ``ARRAY`` and ``MAP`` is supported when the element type of
+    the array is one of the supported types, or when the key type of the map
+    is ``VARCHAR`` and value type of the map is one of the supported types.
+    Behaviors of the casts are shown with the examples below::
+
+        SELECT CAST(NULL AS JSON); -- NULL
+        SELECT CAST(1 AS JSON); -- JSON '1'
+        SELECT CAST(9223372036854775807 AS JSON); -- JSON '9223372036854775807'
+        SELECT CAST('abc' AS JSON); -- JSON '"abc"'
+        SELECT CAST(true AS JSON); -- JSON 'true'
+        SELECT CAST(1.234 AS JSON); -- JSON '1.234'
+        SELECT CAST(ARRAY[1, 23, 456] AS JSON); -- JSON '[1,23,456]'
+        SELECT CAST(ARRAY[1, NULL, 456] AS JSON); -- JSON '[1,null,456]'
+        SELECT CAST(ARRAY[ARRAY[1, 23], ARRAY[456]] AS JSON); -- JSON '[[1,23],[456]]'
+        SELECT CAST(MAP(ARRAY['k1', 'k2', 'k3'], ARRAY[1, 23, 456]) AS JSON); -- JSON '{"k1":1,"k2":23,"k3":456}'
+
+    Note that casting from NULL to ``JSON`` is not straightforward. Casting
+    from a standalone ``NULL`` will produce a SQL ``NULL`` instead of
+    ``JSON 'null'``. However, when casting from arrays or map containing
+    ``NULL``\s, the produced ``JSON`` will have ``null``\s in it.
+
+.. function:: Cast from JSON
+
+    Cast from ``JSON`` to ``BOOLEAN``, ``TINYINT``, ``SMALLINT``, ``INTEGER``,
+    ``BIGINT``, ``REAL``, ``DOUBLE``, ``VARCHAR`` is supported.
+    Cast to ``ARRAY`` and ``MAP`` is supported when the element type of
+    the array is one of the supported types, or when the key type of the map
+    is ``VARCHAR`` and value type of the map is one of the supported types.
+    Behaviors of the casts are shown with the examples below::
+
+        SELECT CAST(JSON 'null' AS VARCHAR); -- NULL
+        SELECT CAST(JSON '1' AS INTEGER); -- 1
+        SELECT CAST(JSON '9223372036854775807' AS BIGINT); -- 9223372036854775807
+        SELECT CAST(JSON '"abc"' AS VARCHAR); -- abc
+        SELECT CAST(JSON 'true' AS BOOLEAN); -- true
+        SELECT CAST(JSON '1.234' AS DOUBLE); -- 1.234
+        SELECT CAST(JSON '[1,23,456]' AS ARRAY(INTEGER)); -- [1, 23, 456]
+        SELECT CAST(JSON '[1,null,456]' AS ARRAY(INTEGER)); -- [1, NULL, 456]
+        SELECT CAST(JSON '[[1,23],[456]]' AS ARRAY(ARRAY(INTEGER))); -- [[1, 23], [456]]
+        SELECT CAST(JSON '{"k1":1,"k2":23,"k3":456}' AS MAP(VARCHAR, INTEGER)); -- {k1=1, k2=23, k3=456}
+
+    JSON arrays can have mixed element types. And JSON maps can have mixed
```

Comment:
"JSON arrays can have mixed element types and JSON maps can have mixed value types."